### PR TITLE
Added a system for multiple controller button icons;

### LINKED ISSUE #35: Suggestion: an option to show Xbox version of console controller key hints
It's much easier to understand for players with Xbox controllers. If the mod could not detect the controller type automatically by itself, a manual option could be added to in-game settings.

### PR DESCRIPTION
**Pull request by FusionLord**
Fixes #35;

### CODE REVIEW

**Comment by MrCrayfish** [comment]
I'm on the edge of whether I merge this or I write from scratch because I will most likely make some big changes to this code if merged making this PR pointless.

**Comment by FusionLord** [comment]
What would you like changed about it? It is Friday night and I have nothing to do. :P

File: src/main/java/com/mrcrayfish/controllable/client/settings/EnumIcons.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    xInput;
+
+    // In-case the user edits the options manually, and enters an invalid value
+    public static EnumIcons valueOfOrDefault(String value) {
+        try {
+            return valueOf(value);
+        } catch (Exception ignored) {
+            return DualShock;
+        }
+    }
```
Comment (MrCrayfish) [inline src/main/java/com/mrcrayfish/controllable/client/settings/EnumIcons.java]: You can now find the code style xml in the project assuming you are using Idea.

File: src/main/java/com/mrcrayfish/controllable/client/settings/ControllerOptions.java [hunk 2]
```diff
@@ -23,6 +23,14 @@
 {
     private static final DecimalFormat FORMAT = new DecimalFormat("0.0#");
 
+    public static final SliderPercentageOption ICONS = new ControllableSliderPercentageOption("controllable.options.icons", 0, EnumIcons.values().length - 1, 1, gameSettings -> {
```
Comment (MrCrayfish) [inline src/main/java/com/mrcrayfish/controllable/client/settings/ControllerOptions.java]: I've added a new class ControllableEnumOption which is better than using a slider to change an enum.

**Comment by MrCrayfish** [comment]
I've now added this in https://github.com/MrCrayfish/Controllable/commit/0cb25f1c9aa839916e73f7d2a2e1e588d731eeb1 using the new ControllableEnumOption

**Pull request closed without merging**